### DISCUSSION TITLE
Backwards compatibility and versioning

### DISCUSSION THREAD

**Discussion by ghost** [Bforartists/Bforartists#2395] [General]
Blender has a system for versioning,
![image](https://user-images.githubusercontent.com/25552173/117528070-a0288300-afd0-11eb-8c69-8161eaefa4c3.png)
when opening old files, we can handle changes in UI, for example panel widths, scrollbar visibility and so on.

So far we have overcome this by updating startup.blend s, and it seems "Load UI" is disabled by default, which overcomes the problem, but doesn't really fix it,

This is a discussion so we are aware of this, and discuss ideas on this

**Comment by ReinerBforartists** [comment]
Hm. I am not really happy to touch this blender versioning. Because this could easily break the compatibility between Blender and Bforartists then.

**Comment by ghost** [comment]
I see, but also sometimes when load UI is on, things are not right..., anyways it is not urgent yet

**Reply by ReinerBforartists** [reply]
Yes i know. For example, the new grid editor uses another internal number in Bforartists. And blend files saved in Blender with this editor open will display the wrong editor in Bforartists then, and vice versa. But i don't see how we could fix this with an own versioning file.

You could make a tracker issue. But for me this is over my limits. I have no idea how to deal with this files.

**Reply by ghost** [reply]
Well I haven't too yet ^^, so we discussed, hopefully in near future we investigate this topic

**Comment by ghost** [comment]
bumping this again, in many instances we had issues regarding default values being stored in startup blend files, we update startup blends, but user blends still have the issue, this little convo helps, we should definitely in future do something about it
![image](https://user-images.githubusercontent.com/25552173/147070657-42da0cb8-12db-4ba4-94bf-4b50b6cc184d.png)

**Comment by ReinerBforartists** [comment]
I am pretty aware of the trouble. But the situation is still the same. We rely at the Blender master. Changing the values will not change anything at this fact, but break a thousand other things then. So please, don't do this, don't just alter the numbers! The versioning numbers exists for good reasons! It might help in one or two cases. But break things in many others. Different versions have api changes for example.

The only thing that could help is to build our release versions from the release versions of Blender. And not from the master. But this would mean to fork Bforartists, since Bforartists relies at the master. And good luck with updating and maintaining this nightmare then ...

We simply don't have the needed manpower to provide a solution to this dilemma here. Sorry :(

**Comment by ghost** [comment]
I agree, I just found the convo thought I would share

**Comment by ReinerBforartists** [comment]
Thanks for sharing :)
